repl/src/main/scala/ammonite/repl/interp/Preprocessor.scala

```diff
@@ -70,7 +92,7 @@ object Preprocessor{
     }
 
     val Expr = Processor{
-      case (name, code, tree) => Output(s"val $name = (\n$code\n)", Seq(pprint(name)))
+      case (name, code, tree) => Output(s"val $name = { () =>\n$code\n}.apply", Seq(pprint(name)))
```

Comment:
Nm, I'm an idiot, this is for wrapping expressions. Could you leave a similar comment as the one you left at the `PatVarDef` definition?